ffcx/codegeneration/lnodes.py

```diff
@@ -327,18 +327,20 @@ def __repr__(self):
 class MultiIndex(LExpr):
     """A multi-index for accessing tensors flattened in memory."""
 
+    precedence = PRECEDENCE.SYMBOL
+
     def __init__(self, symbols: list, sizes: list):
         self.dtype = DataType.INT
         self.sizes = sizes
         self.symbols = [as_lexpr(sym) for sym in symbols]
         for sym in self.symbols:
             assert sym.dtype == DataType.INT
 
-        dim = len(sizes)
-        if dim == 0:
+        self.dim = len(sizes)
```

Comment:
I'd rather not store `self.dim` since it is already the same as `len(self.sizes)`. We could add a method or `@property` to return it:
```
@property
def dim(self):
    return len(self.sizes)
```